src/app/[locale]/game/[id]/page.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import { useTranslations } from 'next-intl';
+import { useQuery } from '@tanstack/react-query';
+import Loader from '@icons/Loader.svg';
+import fetchGame from '@/apiUrls/game';
+import queryKeys from '@/constants/queryKeys';
+import ErrorWrapper from '@/components/errorWrapper';
+
+interface IGamePage {
+  params: { id: number | string };
+}
+
+function GamePage({ params }: IGamePage) {
+  const t = useTranslations();
+
+  const gameId = params.id || '';
+
+  const { isLoading, isError, data } = useQuery({
+    queryKey: queryKeys.fetchGame,
+    queryFn: () => fetchGame(gameId),
+    refetchOnWindowFocus: false,
+    gcTime: 0,
+  });
+
+  if (isLoading) {
+    return (
+      <div className="mt-[20%]">
+        <Loader />
+      </div>
+    );
+  }
+
+  if (isError) {
+    return <ErrorWrapper>{t('errors.An error occurred')}</ErrorWrapper>;
+  }
+
+  if (data.error) {
+    return <ErrorWrapper>{t(`errors.${data.error}`)}</ErrorWrapper>;
+  }
```

Comment:
to musimy zrobić jako jeden blok, możemy dodać jakąś funkcję np `getErrorMessage` do której będziemy przekazywać `error` jeśli error będzie mieć `data.error` to wtedy dajemy to tłumaczenie i będzie nam ładnie tłumaczyć błąd, jeśli nie będzie `data.error` to wtedy będziemy wyświetlać defaultową wiadomość `t('errors.An error occurred')`